cli/command/utils.go

```diff
@@ -90,9 +95,35 @@ func PromptForConfirmation(ins io.Reader, outs io.Writer, message string) bool {
 		ins = streams.NewIn(os.Stdin)
 	}
 
-	reader := bufio.NewReader(ins)
-	answer, _, _ := reader.ReadLine()
-	return strings.ToLower(string(answer)) == "y"
+	result := make(chan bool, 1)
+	// We use a context to cancel the prompt or the signal handler
+	ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
using context.TODO here since we would probably want to use a context received form the `PromptForConfirmation` function parameter.